student-store-ui/src/components/App/App.jsx

```diff
@@ -2,17 +2,164 @@ import * as React from "react"
 import Navbar from "../Navbar/Navbar"
 import Sidebar from "../Sidebar/Sidebar"
 import Home from "../Home/Home"
-import "./App.css"
+import ProductDetail from "../ProductDetail/ProductDetail.jsx"
+import Footer from "../Footer/Footer.jsx"
+import NotFound from "../NotFound/NotFound.jsx"
+import axios from "axios";
+import { useEffect, useState } from 'react';
+import { BrowserRouter, Routes, Route, Link } from 'react-router-dom';
+
+import "./App.css";
+
+const categories = ['All', 'Food', 'Clothing', 'Accessories', 'Tech'];
+const URL = 'https://codepath-store-api.herokuapp.com/store'
+
 
 export default function App() {
+  const [searchStatus, setSearch] = React.useState('')
+  const [categoryStatus, setCategory] = React.useState('All')
+  const [products, setProducts] = React.useState([])
+  const [currentProducts, setCurrentProducts] = React.useState([])
```

Comment:
Another comment on `currentProducts`, does it need to be actual state? Or could we purely derive it from the other state we already have? 

```suggestion
const currentProducts = filterProducts({ products, searchText, categoryStatus })
```

---
*As a bonus step, we could add in [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo)  so we don't re-calculate the list on every render:*
```suggestion
const currentProducts = useMemo(
  () => filterProducts({ products, searchText, categoryStatus }), 
  [products, searchText, categoryStatus]
)
```

---
Thinking about this I actually noticed a bit of a bug with the current code:
* Enter search text
* Select a category filter
* You can observe the search text filter is no longer applied.